File: scrb-introtask/intro_py_pkg/intro_py_pkg/subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String 
from std_msgs.msg import Int8

class IntroSubscriber(Node):
    def __init__(self):
        super().__init__('intro_subscriber')
        # subscribe to String messages on 'string_topic'
        self.string_subscription_ = self.create_subscription(String, 'string_topic', self.string_callback, 5)
        # subscribe to Int8 messages on 'int_topic'
        self.int8_subscription_ = self.create_subscription(Int8, 'int_topic', self.int8_callback, 5)
        
        
    # callback for String messages
    def string_callback(self, strmsg):
        self.get_logger().info(f'i heard string: {strmsg.data}')

    # callback for Int8 messages
    def int8_callback(self, int8msg):
        self.get_logger().info(f'i heard int: {int8msg.data}, f({int8msg.data})={int8msg.data*10}')
        
        

def main(args=None):
    rclpy.init(args=args)
    node = IntroSubscriber() 
    rclpy.spin(node)
    node.destroy_node() 
    rclpy.shutdown()



if __name__ == "__main__":
	main()

File: imu_compass_ws/src/transducer-imu/src/subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
//see publisher.cpp for more comments on the methods (theyre relatively similar)
class IntroSubscriber : public rclcpp::Node
{


	public:
		IntroSubscriber() : Node("intro_subscriber") {//constructor
			//<std_msg...> defines the type	
			//we use the string_topic topic and set the queue/buffer to 5.
						
			string_subscription_ = this->create_subscription<std_msgs::msg::String>(
				"string_topic", 5, std::bind(&IntroSubscriber::callback, this, std::placeholders::_1));
		}
		


	private:
		void callback(const std_msgs::msg::String::SharedPtr msg)
		{
			RCLCPP_INFO(this->get_logger(), "I heard: %s", msg->data.c_str());
		}

		rclcpp::Subscription<std_msgs::msg::String>::SharedPtr string_subscription_;


};
	



int main(int argc, char **argv){

	rclcpp::init(argc, argv);
	rclcpp::spin(std::make_shared<IntroSubscriber>());
	rclcpp::shutdown();
	return 0;
}
